buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy

```diff
@@ -126,12 +126,17 @@ class GuideProjectGenerator implements Closeable {
             }
 
             for (App app: metadata.apps) {
-                List<String> appFeatures = app.features
+                List<String> appFeatures = [] + app.features
```

Comment:
I saw this when removing the mockito feature for Spock - if Groovy is processed before Kotlin or Java, the removal affects those too. This wasn't noticed for GraalVM because the build still compiles and tests pass, but it was causing the Java/Kotlin apps to fail because they didn't have the mockito dependency. Making a copy for each loop iteration makes the change local to that loop iteration and doesn't affect Java or Kotlin, only the current Groovy app